src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,555 @@
+            {
+                // if the number of operators for the quorum is less than or equal to the max operator count, 
+                // then the quorum has not reached the max operator count
+                if(numOperatorsPerQuorum[i] <= operatorSetParam.maxOperatorCount) {
+                    continue;
+                }
+
+                require(
+                    operatorKickParams[i].quorumNumber == quorumNumber, 
+                    "BLSRegistryCoordinatorWithIndices.registerOperatorWithCoordinator: quorumNumber not the same as signed"
+                );
+
+                // get the total stake for the quorum
+                uint96 totalStakeForQuorum = stakeRegistry.getCurrentTotalStakeForQuorum(quorumNumber);
+                bytes32 operatorToKickId = _operators[operatorKickParams[i].operator].operatorId;
+                uint96 operatorToKickStake = stakeRegistry.getCurrentOperatorStakeForQuorum(operatorToKickId, quorumNumber);
+                uint96 registeringOperatorStake = stakeRegistry.getCurrentOperatorStakeForQuorum(operatorIdsToSwap[0], quorumNumber);
+
+                // check the registering operator has more than the kick BIPs of the operator to kick's stake
+                require(
+                    registeringOperatorStake > operatorToKickStake * operatorSetParam.kickBIPsOfOperatorStake / BIPS_DENOMINATOR,
+                    "BLSRegistryCoordinatorWithIndices.registerOperatorWithCoordinator: registering operator has less than kickBIPsOfOperatorStake"
+                );
+                
+                // check the that the operator to kick has less than the kick BIPs of the total stake
+                require(
+                    operatorToKickStake < totalStakeForQuorum * operatorSetParam.kickBIPsOfTotalStake / BIPS_DENOMINATOR,
+                    "BLSRegistryCoordinatorWithIndices.registerOperatorWithCoordinator: operator to kick has more than kickBIPSOfTotalStake"
+                );
+            }
+            
+            // kick the operator
+            _deregisterOperatorWithCoordinator(
+                operatorKickParams[i].operator, 
+                quorumNumbers[i:i+1], 
+                operatorKickParams[i].pubkey, 
+                operatorIdsToSwap
+            );
+        }
+    }
+
+    /**
+     * @notice Deregisters the msg.sender as an operator from the middleware
+     * @param quorumNumbers are the bytes representing the quorum numbers that the operator is registered for
+     * @param deregistrationData is the the data that is decoded to get the operator's deregistration information
+     * @dev `deregistrationData` should be a tuple of the operator's BLS public key, the list of operator ids to swap
+     */
+    function deregisterOperatorWithCoordinator(
+        bytes calldata quorumNumbers,
+        bytes calldata deregistrationData
+    ) external onlyWhenNotPaused(PAUSED_DEREGISTER_OPERATOR) {
```

Comment:
might help to point out or explain somewhere that these overloaded versions of functions with just 'bytes' input data are part of the more general interface / (seemingly) exist so developers can override the function to add more parameters, without changing the function signature?
also curious on thoughts about marking some or all functions in these contracts as 'virtual' to enable easier overriding -- personally feels like it might be overkill / I wouldn't want to encourage this too much, as there would be a lot of potential footguns for developers who somewhat blindly override a function.